config/management/src/error.rs

```diff
@@ -14,20 +14,14 @@ pub enum Error {
     BackendParsingError(String),
     #[error("Invalid arguments: {0}")]
     CommandArgumentError(String),
-    #[error("Local storage unavailable, please check your configuration: {0}")]
-    LocalStorageUnavailable(String),
-    #[error("Failed to read, {0}, from local storage: {1}")]
-    LocalStorageReadError(&'static str, String),
-    #[error("Failed to sign {0} with {1} using local storage: {2}")]
-    LocalStorageSigningError(&'static str, &'static str, String),
-    #[error("Failed to write, {0}, to local storage: {1}")]
-    LocalStorageWriteError(&'static str, String),
-    #[error("Failed to read, {0}, from remote storage: {1}")]
-    RemoteStorageReadError(&'static str, String),
-    #[error("Failed to write, {0}, to remote storage: {1}")]
-    RemoteStorageWriteError(&'static str, String),
-    #[error("Remote storage unavailable, please check your configuration: {0}")]
-    RemoteStorageUnavailable(String),
+    #[error("{0} storage unavailable, please check your configuration: {1}")]
+    StorageUnavailable(&'static str, String),
+    #[error("Failed to read, {1}, from {0} storage: {2}")]
+    StorageReadError(&'static str, &'static str, String),
+    #[error("Failed to sign {1} with {2} using {0} storage: {2}")]
+    StorageSigningError(&'static str, &'static str, &'static str, String),
+    #[error("Failed to write, {1}, to {0} storage: {2}")]
+    StorageWriteError(&'static str, &'static str, String),
```

Comment:
tiny nit: it seems odd to me that we're using {1}, then {0}, then {2}? Perhaps we can change the call sites so that we can just using the arguments in order? e.g., Error::StorageReadError(key_name, storage_name, e.to_string()) still makes sense?

Reply:
I personally wanted the types passed in to be indicative of their status... the first one tells you whom or what is the source, the next one the actor or retrieved piece, and the 3rd is the extra info. I agree it is weird. This is where using named types would be better but also ... uglier... Rust :D